logging: consistent error attribute when emitted through logr

giorio94 opened the pull request:
Extend the log attributes mangling logic to ensure consistency in the attribute key used to identify an error. Indeed, the `logr.Error` method takes an error as first parameter, and the slog adapter converts it into an attribute identified as `err`. Let's replace that into `error` to ensure consistency with all the other occurrences.

giorio94 committed:
logging: consistent error attribute when emitted through logr

Extend the log attributes mangling logic to ensure consistency in the
attribute key used to identify an error. Indeed, the logr.Error method
takes an error as first parameter, and the slog adapter converts it into
an attribute identified as `err`. Let's replace that into `error` to
ensure consistency with all the other occurrences.

Related: d91e387bdbb6 ("cec: Switch to slog for CEC")
Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

mhofstetter commented on pkg/logging/slog.go:
```diff
@@ -95,6 +100,11 @@ func replaceLevel(groups []string, a slog.Attr) slog.Attr {
 			Key:   a.Key,
 			Value: slog.StringValue(strings.ToLower(a.Value.String())),
 		}
+	case logrErrorKey:
```
the name of this function (`replaceLevel`) is no longer correct (probably already before this change).

giorio94 replied:
Yeah, makes sense. I've generalized the two function names, and made one build on top of the other to ensure consistency and reduce code duplication.

mhofstetter commented on pkg/logging/slog.go:
```diff
@@ -110,6 +120,11 @@ func replaceLevelAndDropTime(groups []string, a slog.Attr) slog.Attr {
 			Key:   a.Key,
 			Value: slog.StringValue(strings.ToLower(a.Value.String())),
 		}
+	case logrErrorKey:
```
the name of this function (`replaceLevelAndDropTime`) is no longer correct (probably already before this change).

mhofstetter reviewed (commented):
Thanks! Only two questions regarding the naming of the "replace"-functions. Maybe it's worth to adjust them to something more generic? (`replaceFuncWithoutTimestamp` & `replaceFunc` ?)

giorio94 committed:
logging: slight generalization of the replace attr functions

Slightly generalize the log attributes mangling logic in preparation for
the subsequent extension, changing the function names to be agnostic of
the specific operations and making one rely on the other, to ensure
consistency and reduce code duplication.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

mhofstetter reviewed (approved):
LGTM, thanks!
